### PR TITLE
Changing array sizing datatype to size_t

### LINKED ISSUE #372: Increase array sizing datatype to size_t
Currently RAJAPerf is using int to pass around size information for arrays on host (for allocation/init purposes). This gives an upper bound on some of the tests in terms of how large of a problem size we can safely run. For example, the Basic_ARRAY_OF_PTRS benchmark allocates an array that is 26x the problem size. If we try to run the benchmark with the inputs:
```bash
$ ./raja-perf.exe  -v Base_HIP --size 5e7 -sp --disable-warmup -k Basic_ARRAY_OF_PTRS


Reading command line input...

Setting up suite based on input...


RAJA performance suite run summary....
--------------------------------------

How suite will be run:
         # passes = 1
         Kernel size = 5e+07
         Kernel rep factor = 1
         Output files will be named ./RAJAPerf*

The following kernels and variants (when available for a kernel) will be run:

Data Spaces
--------
Seq - Host
OpenMP - Omp
Hip - HipDevice

Variants and Tunings
--------
Base_HIP-block_256

Kernels               ,    Problem size ,    Reps ,    Iterations/rep , Kernels/rep ,     Bytes/rep ,     FLOPS/rep
-------------------------------------------------------------------------------------------------------------------
Basic_ARRAY_OF_PTRS   ,        50000000 ,      50 ,          50000000 ,           1 ,   10800000000 ,    1300000000


Running specified kernels and variants...

Pass through suite # 0

Run kernel -- Basic_ARRAY_OF_PTRS
        Running Base_HIP variant
Segmentation fault (core dumped)
```

Passing this through a debugger shows:
```bash
Thread 1 "raja-perf.exe" received signal SIGSEGV, Segmentation fault.
0x00007ffff7635f48 in rajaperf::detail::initData (ptr=@0x355dd8: 0x7ffb2b600000, len=len@entry=1300000000)
    at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/common/DataUtils.cpp:404
404         ptr[i] = factor*(i + 1.1)/(i + 1.12345);
(gdb) bt
#0  0x00007ffff7635f48 in rajaperf::detail::initData (ptr=@0x355dd8: 0x7ffb2b600000, len=len@entry=1300000000)
    at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/common/DataUtils.cpp:404
#1  0x00007ffff7d7122c in rajaperf::allocAndInitData<double> (dataSpace=rajaperf::DataSpace::HipDevice, ptr=@0x355dd8: 0x7ffb2b600000, len=1300000000,
    align=64) at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/./common/DataUtils.hpp:301
#2  0x00007ffff7644547 in rajaperf::KernelBase::execute (this=0x3554a0, vid=rajaperf::Base_HIP, tune_idx=0)
    at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/common/KernelBase.cpp:251
#3  0x00007ffff7639457 in rajaperf::Executor::runKernel (this=this@entry=0x7fffffffd268, kernel=0x3554a0, print_kernel_name=<optimized out>)
    at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/common/Executor.cpp:610
#4  0x00007ffff7638b2e in rajaperf::Executor::runSuite (this=0x7fffffffd268)
    at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/common/Executor.cpp:563
#5  0x0000000000201f60 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/semiller/software/raja/RAJAPerfTools/RAJAPerf/src/RAJAPerfSuiteDriver.cpp:46
(gdb) list
399     void initData(Real_ptr& ptr, int len)
400     {
401       Real_type factor = ( data_init_count % 2 ? 0.1 : 0.2 );
402
403       for (int i = 0; i < len; ++i) {
404         ptr[i] = factor*(i + 1.1)/(i + 1.12345);
405       }
406
407       incDataInitCount();
408     }
(gdb) p len
$1 = 1300000000
(gdb) p i
$2 = 0
(gdb) p ptr
$3 = (rajaperf::Real_ptr &) @0x355dd8: 0x7ffb2b600000
```

Some thoughts:
- It may be a good idea to check if the malloc'ed pointer is valid in the allocData call.
  - It wouldn't have help here since the overflow ended up positive.
- Changing all the indexing globally to use size_t instead of int is a solution, but it will likely degrade performance in the kernels.
  - Should it be limited to host/setup calls?
  - Should the index type be chosen at compile time (like the `Real` type)?

### PR DESCRIPTION
**Pull request by seanofthemillers**
# Increasing size type for arrays

- This PR is a bugfix/feature
- It does the following:
   - Introduces Size_type (currently size_t)
   - Changes `len` (array sizing) parameters from int to Size_type
   - Closes #372 
   - Allows for running benchmarks at larger problem sizes

Do not merge this PR. Please review/approve https://github.com/LLNL/RAJAPerf/pull/374 for merging

### CODE REVIEW

**Comment by seanofthemillers** [comment]
Still testing

**Comment by seanofthemillers** [comment]
Tests:
```bash
./raja-perf.exe  -v Base_HIP RAJA_HIP --size 5e7 -sp -ek Polybench_JACOBI_2D Polybench_FDTD_2D
```
runs to completion with these changes. 

Without the fixes, the run fails during the warmup kernels with:
```bash
Run kernel -- Algorithm_SORT
        No Base_HIP variant
        Running RAJA_HIP variant
terminate called after throwing an instance of 'std::runtime_error'
  what():  HIPassert: out of memory /home/semiller/software/raja/RAJAPerf/src/common/HipDataUtils.hpp:159
Aborted (core dumped)
```

File: src/common/DataUtils.cpp [hunk 1]
```diff
@@ -132,7 +132,7 @@ void copyHostData(void* dst_ptr, const void* src_ptr, size_t len)
 /*
  * Allocate data arrays of given type.
  */
-void* allocHostData(size_t len, size_t align)
+void* allocHostData(Size_type len, size_t align)
```
Comment (MrBurmark) [inline src/common/DataUtils.cpp:135]: As long as we're changing this we should also change align.

**Comment by MrBurmark** [comment]
@rhornung67 what do you think of a Size_type alias vs using size_t? 
If we go with an alias then we should also get the cuda and hip allocation methods.

**Comment by rhornung67** [comment]
@seanofthemillers following what we have for Corbin, I'm making a request to add you to the LLNL organization on GitHub. After t hat goes through, I will add you to the RAJA-contrib team. Then, you will be able to create PRs in the RAJA and RAJAPerf projects directly rather than on a fork. Due to security policies, we cannot run GItLab CI on LC machines for PRs from forks. We require all checks to pass before merging.

I will pull you PR branch into our repo and set up a new PR for that.

**Comment by seanofthemillers** [comment]
Thanks @rhornung67, I believe I am now added to the LLNL org and am ready to be added to the RAJA teams.

**Comment by rhornung67** [comment]
@seanofthemillers i added you to the team. i will pull in your fork branch and set up a new PR x-linking this one.

**Comment by rhornung67** [comment]
@seanofthemillers the PR from a branch in this repo is https://github.com/LLNL/RAJAPerf/pull/374

**Pull request merged**